test/bin/start_api_test_server.sh

```diff
@@ -33,17 +33,67 @@ _get_osmt_project_dir() {
   echo "${project_dir}"
 }
 
-project_dir="$(_get_osmt_project_dir)" || exit 135
+curl_with_retry() {
+  local -i rc=-1
+  local -i retry_limit=12
+  until [ ${rc} -eq 0 ] && [ ${retry_limit} -eq 0 ]; do
+      echo_info "Attempting to request the index page of the OSMT Spring app with curl..."
+      curl -s "${BASE_URL}" 1>/dev/null 2>/dev/null
+      rc=$?
+      if [[ ${rc} -eq 0 ]]; then
+        echo_info "Index page loaded. Proceeding..."
+        return 0
+      fi
+      if [[ ${retry_limit} -eq 0 ]]; then
+        echo_err "Could not load the index page."
+        return 1
+      fi
+      if [[ ${rc} -ne 0 ]]; then
+        echo_info "Could not load the index page. Will retry ${retry_limit} more times. Retrying in 10 seconds..."
+      fi
+      let retry_limit-=1
+      sleep 10
+  done
+}
+
+
+echo_info() {
+  echo "INFO: $*"
+}
+
+echo_err() {
+  echo "ERROR: $*" 1>&2;
+}
 
-# start the API test  Docker compose stack and Spring app server
-"${project_dir}/osmt_cli.sh" -s
+error_handler() {
+  echo_err "Trapping at error_handler. Exiting"
+  # clean up API test Docker resources
+  "${project_dir}/osmt_cli.sh" -e || exit 135
+}
 
-# load CI static dataset
-#"${project_dir}/osmt_cli.sh" -l
+main() {
+  local project_dir; project_dir="$(_get_osmt_project_dir)" || exit 135
+  local logFile; logFile="${project_dir}/tmp/osmt_spring_app.log"
 
-# reindex ElasticSearch
-#"${project_dir}/osmt_cli.sh" -r
+  # start the API test Docker compose stack and Spring app server, detached
+  echo_info "Starting OSMT Spring app for ${OSMT_STACK_NAME}. Output is suppressed, because console is detached."
+  echo_info "See 'osmt_spring_app.log' for console output. Proceeding..."
+  # "${project_dir}/osmt_cli.sh" -s  1>/dev/null 2>/dev/null & disown  || exit 135
+  touch logFile
+  "${project_dir}/osmt_cli.sh" -s  1>logFile 2>logFile & disown  || exit 135
```

Comment:
If you want the value of the variable, you need $log_file.

If you want proper expansion of any values in $log_file, then you need ${log_file}.

If you want to ensure that you actually have a value even if it's just an empty string, then you want "${log_file}".

shellcheck, shellcheck, shellcheck... :)